### ISSUE TITLE
Not getting binary mask using model.predict

### ISSUE THREAD

**Issue by ad12** [bonlime/keras-deeplab-v3-plus#28]
I trained the model using model.fit_generator after preprocessing my input data using the ```model.preprocess_input``` function. 

I save the weights to a checkpoint file and load them back using ```model.load_weights(path_to_weights)```. Now the model should be ready for testing. I run the analysis on a single test image that has also been preprocessed using ```y_pred = model.predict(test_image)```. However, y_pred is not a binary mask.

Do we have to add our own output activation layer to the model? Or any other thoughts on why this may be occuring? Thanks!

**Comment by jimmy15923** [comment]
Do we need to add Activation function like softmax or sigmoid by ourself? 
Since the model will output logits in last layer?

**Comment by zongzi12321** [comment]
I loaded the model with keras wights , then I preprocessesd the image scaled to [-1.,1.]. and added softmax layer at the top, However the output is not a mask. Every channel of the output is like this:
the code is as follow: ![image](https://wx2.sinaimg.cn/mw690/86263d9cgy1fsirso01q5j20e80e842o.jpg)
`deeplab_model = Deeplabv3(input_shape=(512,512,3), classes=4, weights='pascal_voc',backbone='xception', OS=8)
reshape=Reshape((512*512,4))(deeplab_model.output)
activation=Activation('softmax')(reshape)
predict=Reshape((512,512,4))(activation)
model = Model(inputs=deeplab_model.input, outputs=predict)

img = cv2.imread("image1.jpg")
img = cv2.resize(img, (512, 512))
img = img.reshape(-1, 512, 512, 3)
img=img/127.5-1
result = model.predict(img)
qq=result[0].reshape(4,512,512)
cv2.imwrite('1.jpg',qq[0]*255)
`


I also tried not to reshape the image to((512*512,4)),the result is still the same.  What should I do? 
Thanks a lot!
